metatheory/default.nix

```diff
@@ -0,0 +1,49 @@
+{ pkgs, cleanSourceHaskell, haskellPackages, agda, AgdaStdlib }:
+let 
+  # The agda builder doesn't work properly with library files, we need to use direct include flags
+  libFilter = name: type: let basename = baseNameOf (toString name); in !(pkgs.lib.hasSuffix ".agda-lib" basename);
+  # Most of the filters for Haskell source are good for us too
+  cleanSourceAgda = src: pkgs.lib.cleanSourceWith { filter = libFilter; src = (cleanSourceHaskell src); };
+in rec { 
+  plutus-formal = agda.mkDerivation (self: rec {
+    name = "plutus-formal";
+
+    buildDepends = [ AgdaStdlib ];
+
+    # We can't just add more flags, annoyingly, so we have to repeat some of the existing flags.
+    # Passing the html output flags gets us the literate output, and still checks the Agda, so this
+    # derivation can do double duty.
+    buildFlags = pkgs.lib.concatStringsSep " " (["--html" "--html-highlight=auto" ] ++ (map (x: "-i " + x) self.includeDirs));
```

Comment:
does this mean it's easier to pull in the std-lib directly `-i` rather than using the agda library management stuff?

Reply:
The nixpkgs agda builder doesn't support library stuff. I've opened an issue upstream - it's fairly unloved. Trying not to get sucked into writing a new one :grimacing: